test/e2e/apimachinery/crd_publish_openapi.go

```diff
@@ -87,28 +90,36 @@ var _ = SIGDescribe("CustomResourcePublishOpenAPI [Privileged:ClusterAdmin]", fu
 			framework.Failf("failed to delete valid CR: %v", err)
 		}
 
-		ginkgo.By("client-side validation (kubectl create and apply) rejects request with value outside defined enum values")
+		ginkgo.By("server-side validation (kubectl create and apply) rejects request with value outside defined enum values")
 		badEnumValueCR := fmt.Sprintf(`{%s,"spec":{"bars":[{"name":"test-bar", "feeling":"NonExistentValue"}]}}`, meta)
 		if _, err := framework.RunKubectlInput(f.Namespace.Name, badEnumValueCR, ns, "create", "-f", "-"); err == nil || !strings.Contains(err.Error(), `Unsupported value: "NonExistentValue"`) {
 			framework.Failf("unexpected no error when creating CR with unknown enum value: %v", err)
 		}
 
-		ginkgo.By("client-side validation (kubectl create and apply) rejects request with unknown properties when disallowed by the schema")
+		ginkgo.By("server-side validation (kubectl create and apply) rejects request with unknown properties when disallowed by the schema")
 		unknownCR := fmt.Sprintf(`{%s,"spec":{"foo":true}}`, meta)
-		if _, err := framework.RunKubectlInput(f.Namespace.Name, unknownCR, ns, "create", "-f", "-"); err == nil || !strings.Contains(err.Error(), `unknown field "foo"`) {
-			framework.Failf("unexpected no error when creating CR with unknown field: %v", err)
+		expectedErr := `unknown field "spec.foo"`
+		if !utilfeature.DefaultFeatureGate.Enabled(features.ServerSideFieldValidation) {
```

Comment:
feature gate checks in e2e doesn't work. you have to detect enablement on the server or tolerate both enabled/disabled behavior

Reply:
modified to tolerate both enabled/disabled behavior